reactos/drivers/usb/usbhub_new/usbhub.c

```diff
@@ -3347,86 +3329,75 @@ USBH_FdoIdleNotificationCallback(IN PVOID Context)
                           FALSE,
                           NULL);
 
-    Port = 0;
+    PortData = HubExtension->PortData;
 
-    if (HubExtension->HubDescriptor->bNumberOfPorts == 0)
+    for (Port = 0;
+         Port < HubExtension->HubDescriptor->bNumberOfPorts;
+         Port++)
     {
         if ((HubExtension->HubFlags & USBHUB_FDO_FLAG_DEVICE_STOPPING) ||
             (USBH_CheckIdleAbort(HubExtension, FALSE, FALSE) != TRUE))
         {
+            IsReady = FALSE;
             goto IdleHub;
         }
-    }
-    else
-    {
-        PortData = HubExtension->PortData;
-
-        while (TRUE)
-        {
-            PortDevice = PortData[Port].DeviceObject;
-
-            if (PortDevice)
-            {
-                PortExtension = PortDevice->DeviceExtension;
 
-                IdleIrp = PortExtension->IdleNotificationIrp;
+        PortDevice = PortData[Port].DeviceObject;
 
-                if (!IdleIrp)
-                {
-                    break;
-                }
-
-                IoStack = IoGetCurrentIrpStackLocation(IdleIrp);
+        if (PortDevice)
+        {
+            PortExtension = PortDevice->DeviceExtension;
 
-                CallbackInfo = IoStack->Parameters.DeviceIoControl.Type3InputBuffer;
+            IdleIrp = PortExtension->IdleNotificationIrp;
 
-                if (!CallbackInfo)
-                {
-                    break;
-                }
+            if (!IdleIrp)
+            {
+                break;
```

Comment:
usbhub.c:3355: before the latest change, these "break" cases inside the loop used to set IsReady to false, then fall through to the IdleHub label. So I think they should probably say `IsReady = FALSE; goto IdleHub;` now?